src/platform_impl/macos/view.rs

```diff
@@ -890,6 +911,8 @@ fn mouse_motion(this: &Object, event: id) {
             return;
         }
 
+        update_potentially_stale_modifiers(state, event);
```

Comment:
Why have you decided to add this on `key_up` and `mouse_motion`?

I think `key_up` shouldn't be super critical, but I'd imagine the less we do on `mouse_motion` the better.

Reply:
I added this just for consistency's sake, but I did also have reservations about adding it on `mouse_motion`. I'd be happy to remove it from one or both locations, what do you think?

Reply:
I'd only add it to scenarios where it can actually trigger a relevant bug. I'm not sure about a scenario with mouse motion where it could trigger an incorrect modifier state.

But maybe consistency isn't that bad either, but it's generally a very ugly solution imho.

Reply:
Alright, I'll also add removing it with `mouse_motion` but I'll keep it in `key_up` for now.